airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -213,65 +148,6 @@ def _stream_configs(self, manifest: Mapping[str, Any]):
-    def expand_schema_interfaces(expand_class: type, visited: dict) -> type:
-        """
-        Recursive function that takes in class type that will have its interface fields unpacked and expended and then recursively
-        attempt the same expansion on all the class' underlying fields that are declarative component. It also performs expansion
-        with respect to interfaces that are contained within generic data types.
-        :param expand_class: The declarative component class that will have its interface fields expanded
-        :param visited: cache used to store a record of already visited declarative classes that have already been seen
-        :return: The expanded declarative component
-        """
-
-        # Recursive base case to stop recursion if we have already expanded an interface in case of cyclical components
-        # like CompositeErrorHandler
-        if expand_class.__name__ in visited:
-            return visited[expand_class.__name__]
-        visited[expand_class.__name__] = expand_class
-
-        next_classes = []
-        class_fields = fields(expand_class)
-        for field in class_fields:
-            unpacked_field_types = DeclarativeComponentFactory.unpack(field.type)
-            expand_class.__annotations__[field.name] = unpacked_field_types
-            next_classes.extend(ManifestDeclarativeSource._get_next_expand_classes(field.type))
-        for next_class in next_classes:
-            ManifestDeclarativeSource.expand_schema_interfaces(next_class, visited)
-        return expand_class
-
-    @staticmethod
-    def _get_next_expand_classes(field_type) -> list[type]:
-        """
-        Parses through a given field type and assembles a list of all underlying declarative components. For a concrete declarative class
-        it will return itself. For a declarative interface it will return its subclasses. For declarative components in a generic type
-        it will return the unpacked classes. Any non-declarative types will be skipped.
-        :param field_type: A field type that
-        :return:
-        """
-        generic_type = typing.get_origin(field_type)
-        if generic_type is None:
-            # We can only continue parsing declarative that inherit from the JsonSchemaMixin class because it is used
-            # to generate the final json schema
-            if inspect.isclass(field_type) and issubclass(field_type, JsonSchemaMixin) and not isinstance(field_type, EnumMeta):
-                subclasses = field_type.__subclasses__()
-                if subclasses:
-                    return subclasses
-                else:
-                    return [field_type]
-        elif generic_type == list or generic_type == Union:
-            next_classes = []
-            for underlying_type in typing.get_args(field_type):
-                next_classes.extend(ManifestDeclarativeSource._get_next_expand_classes(underlying_type))
-            return next_classes
-        return []
-
```

Comment:
I believe the `SchemaEncoder` class below is also only used by the method to `generate_schema()` we're deleting so that can get removed too. Just commenting here since I can't comment on unmodified lines